PROTOCOL.md

```diff
@@ -515,11 +568,33 @@ To achieve the requirements above, related actions from different delta files ne
  - To replay the log, scan all file actions and keep only the newest reference for each logical file.
  - `add` actions in the result identify logical files currently present in the table (for queries). `remove` actions in the result identify tombstones of logical files no longer present in the table (for VACUUM).
 
+# Table Features
+Table Features must only exist on tables that have a supported protocol version. When the table's Reader Version is 3, `readerFeatures` must exist in the `protocol` action, and when the Writer Version is 7, `writerFeatures` must exist in the `protocol` action. `readerFeatures` and `writerFeatures` define the features that readers and writers must implement in order to read and write this table.
+
+Readers and writers must not ignore Table Features when they are present:
+ - to read a table, readers must implement and respect all features listed in `readerFeatures`;
+ - to write a table, writers must implement and respect all features listed in `writerFeatures`. Because writers have to read the table (or only the Delta log) before write, they must implement and respect reader features as well.
+
+## Table Features for new and existing tables
+It is possible to enable Table Features during table creation or on existing tables. The enablement can be only for readers or both readers and writers.  When a new table is created with a Reader Version up to 2 and Writer Version 7, its `protocol` action must only contain `writerFeatures`. When a new table is created with Reader Version 3 and Writer Version 7, its `protocol` action must contain both `readerFeatures` and `writerFeatures`. Creating a table with a Reader Version 3 and Writer Version up to 6 is not allowed.
+
+When upgrading an existing table to Reader Version 3 and/or Writer Version 7, the client should, on a best effort basis, determine which features supported by the original protocol version are used in any historical version of the table, and add only used features to Table Features. The client must assume a feature has been used, unless it can prove that the feature is *definitely* not used in any historical version of the table that is reachable by time travel.
```

Comment:
> the client should, on a best effort basis, determine which features supported by the original protocol version are used in any historical version of the table, and add only used features to Table Features.

This might be a little abstract for readers. Perhaps we could give an example?

e.g. what does it mean or look like when a feature in the original protocol is enabled but not used?

Reply:
Yep, example is added!